### PR TITLE
Feature/side nav mobile

### PR DESCRIPTION
**Pull request by Hariharan482**

### CODE REVIEW

File: src/app/mixins.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@function calculateRem($size) {
+  $remSize: $size / 16px;
+  @return $remSize * 1rem;
+}
+
+@mixin font-size($size) {
```
Comment (laxman1522) [inline src/app/mixins.scss:6]: @Hariharan482 why this mixin has same property twice

File: src/components/SideNavigation/SideNavigation.scss [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+@import '../../app/layout.scss'; // responsive grid and media queries
+
+/* -------------------------------- 
+
+Primary style
+
+-------------------------------- */
+
+// html * {
```
Comment (laxman1522) [inline src/components/SideNavigation/SideNavigation.scss:13]: @Hariharan482 Kindly remove unused code

File: src/components/SideNavigation/SideNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import HomePage from '../HomePage/HomePage';
+import AboutPage from '../AboutPage/AboutPage';
+
+//NOTE: For Mobile View style - Have to refresh screen
+export default function SideNavigation() {
+  const eventsPageRef = useRef(null);
+  const homePageRef = useRef(null);
+  const aboutPageRef = useRef(null);
+
+  const sections = [
```
Comment (laxman1522) [inline src/components/SideNavigation/SideNavigation.tsx:14]: @Hariharan482 Kindly move this to constants

File: src/components/SideNavigation/SideNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  ];
+
+  const rows = new Array(3).fill(null);
+  const boxes = new Array(3).fill(null);
+
+  return (
+    <div id="nav-wrapper">
+      <nav id="cd-vertical-nav">
+        <ul>
+          {sections.map((section) => (
```
Comment (laxman1522) [inline src/components/SideNavigation/SideNavigation.tsx:46]: @Hariharan482  Kindly use null check here sections?.map and kindly assign this to a variable and use it in the return function

File: src/components/SideNavigation/SideNavigation.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      <a className="cd-nav-trigger cd-img-replace">
+        {rows.map((_, rowIndex) => (
+          <span className="row" key={`row-${rowIndex}`}>
+            {boxes.map((_, boxIndex) => (
+              <span className="box" key={`box-${boxIndex}`}></span>
+            ))}
+          </span>
+        ))}
+
+        {/* <span ></span> */}
```
Comment (laxman1522) [inline src/components/SideNavigation/SideNavigation.tsx:70]: @Hariharan482 Kindly remove unused codes hari

File: src/components/EventsPage/EventsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React, { forwardRef } from 'react';
+
+interface EventsPageProps {
+  // You can add any props that the EventsPage component might need in the future
+}
+const EventsPage = forwardRef<HTMLDivElement, EventsPageProps>((props, ref) => {
+  return (
+    <div ref={ref} id="events-section" className=" h-screen">
```
Comment (laxman1522) [inline src/components/EventsPage/EventsPage.tsx:8]: @Hariharan482 Id used here is same as id used in the sideNavigation is this expected ?

File: src/components/AboutPage/AboutPage.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React, { forwardRef } from 'react';
+
+interface AboutPageProps {
+  // You can add any props that the AboutPage component might need in the future
+}
+const AboutPage = forwardRef<HTMLDivElement, AboutPageProps>((props, ref) => {
+  return (
+    <div ref={ref} id="events-section" className=" h-screen">
```
Comment (laxman1522) [inline src/components/AboutPage/AboutPage.tsx:8]: @Hariharan482 id used here is events-section but it is about page

File: src/app/mixins.scss (hunk 1 above)
Comment (Hariharan482) [reply src/app/mixins.scss:6]: removed duplicate property

File: src/components/SideNavigation/SideNavigation.scss (hunk 2 above)
Comment (Hariharan482) [reply src/components/SideNavigation/SideNavigation.scss:13]: removed unused code

File: src/components/SideNavigation/SideNavigation.tsx (hunk 4 above)
Comment (Hariharan482) [reply src/components/SideNavigation/SideNavigation.tsx:46]: added null check

File: src/components/SideNavigation/SideNavigation.tsx (hunk 5 above)
Comment (Hariharan482) [reply src/components/SideNavigation/SideNavigation.tsx:70]: removed unused code

File: src/components/AboutPage/AboutPage.tsx (hunk 7 above)
Comment (Hariharan482) [reply src/components/AboutPage/AboutPage.tsx:8]: changed it to about-section

File: src/components/SideNavigation/SideNavigation.tsx (hunk 3 above)
Comment (Hariharan482) [reply src/components/SideNavigation/SideNavigation.tsx:14]: moved it to constants and created a util to create a ref and assign it to each section

**Pull request merged**